src/main/java/org/opensearch/knn/index/codec/transfer/VectorTransfer.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.transfer;
+
+import lombok.Data;
+import org.opensearch.knn.index.codec.util.SerializationMode;
+
+import java.io.ByteArrayInputStream;
+
+/**
+ * Abstract class to transfer vector value from Java to native memory
+ */
+@Data
+public abstract class VectorTransfer {
```

Comment:
Why this class has to be abstract , I don't see any common functionality implemented. Can we have interface?

Reply:
Same answer above.